src/parsers/espresso.js

```diff
@@ -25,6 +30,468 @@ function toEspressoFormat(materialOrConfig) {
+                        const [key, value] = line.split("=").map((str) => str.trim());
+                        // convert Fortran boolean to JavaScript boolean
+                        if (value === ".true.") {
+                            blockContent[key] = true;
+                        } else if (value === ".false.") {
+                            blockContent[key] = false;
+                        } else if (Number.isNaN(Number(value))) {
+                            // if value is not a number, keep it as a string
+                            blockContent[key] = value.replace(/'/g, ""); // remove single quotes
+                        } else {
+                            blockContent[key] = Number(value); // convert to number
+                        }
+                    }
+                }
+            }
+            result[blockName.slice(1)] = blockContent; // remove '&' from block name
+        } else {
+            result.cards = block;
+        }
+    });
+
+    return result;
+}
+
+/**
+ * @summary Read unit cell parameters from CELL_PARAMETERS card
+ * @param {String} cards
+ * @param {Number} alat
+ * @returns {cell: Number[][], alat: Number, units: String}
+ */
+function getCellParameters(cards, alat = null) {
+    const cellParamsRegex = /CELL_PARAMETERS\s\(([\w]+)\)(?:\n?([!#].*)?)\n(((-?\d+.\d+)\s+)*)/gm;
+    // Extract cell parameters section
+    const cellParamsMatch = cellParamsRegex.exec(cards);
+
+    // Extract the units
+    const units = cellParamsMatch[1];
+    let cellUnits = 1.0;
+
+    if (units === "bohr") {
+        if (alat !== null) {
+            throw new Error(
+                "Lattice parameters given in &SYSTEM celldm/A and CELL_PARAMETERS bohr",
+            );
+        }
+        cellUnits = coefficients.BOHR_TO_ANGSTROM;
+    } else if (units === "angstrom") {
+        if (alat !== null) {
+            throw new Error(
+                "Lattice parameters given in &SYSTEM celldm/A and CELL_PARAMETERS angstrom",
+            );
+        }
+        cellUnits = 1.0;
+    } else if (units === "alat") {
+        if (alat === null) {
+            throw new Error("Lattice parameters must be set in &SYSTEM for alat units");
+        } else cellUnits = alat * coefficients.BOHR_TO_ANGSTROM;
+    } else if (alat === null) {
+        cellUnits = coefficients.BOHR_TO_ANGSTROM;
+    } else {
+        cellUnits = alat;
+    }
+
+    const cellParamsData = cellParamsMatch[3].trim().split("\n");
+
+    const cellVectors = [];
+    cellParamsData.forEach((line) => {
+        const values = line
+            .trim()
+            .match(/([\d.-]+)/g)
+            .map(Number)
+            .map((value) => value * cellUnits);
+        cellVectors.push(values);
+    });
+
+    // eslint-disable-next-line no-param-reassign
+    alat = 1; // TODO: remove and change this
+    return {
+        cell: cellVectors,
+        alat,
+        units,
+    };
+}
+
+/**
+ * @summary Creates cell from ibrav and celldm(i) parameters
```

Comment:
1. Let's include a link to the QE documentation where this is explained pls
2. We should include tests for this function specifically, it's pretty complicated
3. It's not a huge deal for this specific case, but in JS we use `camelCase` not `snake_case` - see how we deal with this in reciprocal cell symmetry points definitions pls.

Reply:
1. ok
2. tests will be in follow up task
3. I don't see a snake_case usage here